tutorials/backend/hasura-authentication/tutorial-site/content/integrations/casdoor.md

```diff
@@ -0,0 +1,149 @@
+2. Add a redirect url: `http://CASDOOR_HOSTNAME/login`
+   ![Casdoor Application Setting](https://github.com/RanTao123/image/blob/main/Casdoor%20Application%20Setting.png?raw=true)
+3. Copy the client ID, we will need it in the following steps.
+
+## Add user in Casdoor
+
+Now you have the application running, but you must create a user and assign them a role next.
+
+Go to the “Users” page and click “Add user” in the top right corner. This will open a new page where you can add the
+user.
+
+![Pic showing the users page](https://github.com/RanTao123/image/blob/main/user.png?raw=true)
+
+Save the user after adding a username and the organization Hasura (other details are optional).
+
+Now you need to set up a password for your user, which you can do by clicking "Manage Your Password."
+
+Choose a password for your user and confirm it.
+
+## Create a Hasura instance
+
+Create a Hasura instance using either
+[Hasura Cloud](https://hasura.io/docs/latest/getting-started/getting-started-cloud/) or
+[Docker](https://hasura.io/docs/latest/getting-started/docker-simple/).
+
+Now, create a `users` table with the following columns:
+
+- `id` of type Text (Primary Key)
+- `username` of type Text
+
+See the image below for reference.
+
+![Picture showing how to create a table in Hasura](https://graphql-engine-cdn.hasura.io/learn-hasura/assets/graphql-hasura-authentication/keycloak/hasura-create-table.png)
+
+The next step is to create a `user` role for the app. Users should be able to see only their records, but not the other
+people’s records.
+
+Configure the `user` role as shown in the image below. For more information, read about
+[configuring permission rules in Hasura](https://hasura.io/docs/latest/graphql/core/auth/authorization/permission-rules/).
+
+![Picture showing how to set permissions in Hasura](https://graphql-engine-cdn.hasura.io/learn-hasura/assets/graphql-hasura-authentication/keycloak/hasura-set-permissions.png)
+
+This way, users can only access their own records.
+
+For testing purposes, add a dummy user. This is to ensure that when you use the JWT token, you only see your user’s
+details and not other users’ details.
+
+![Picture showing how to add a table record in Hasura](https://graphql-engine-cdn.hasura.io/learn-hasura/assets/graphql-hasura-authentication/keycloak/hasura-dummy-user.png)
+
+Next, set the `JWT_SECRET` in Hasura.
+
+## Configure Hasura with Casdoor
+
+In this step, you need to add the **HASURA_GRAPHQL_JWT_SECRET** to Hasura.
+
+To do so, go to the Hasura docker-compose.yaml and then add the new `HASURA_GRAPHQL_JWT_SECRET` as below.
+
+The `HASURA_GRAPHQL_JWT_SECRET` should be in the following format:
+
+```
+HASURA_GRAPHQL_JWT_SECRET: '{"claims_map": {
+      "x-hasura-allowed-roles": ["user","editor"],
+      "x-hasura-default-role": "user",
+      "x-hasura-user-id": "userID"
+    },"jwk_url":"https://door.casdoor.com/.well-known/jwks"}'
+```
+
+Save the change, and restart the Docker container.
```

Comment:
Earlier, you mention that Hasura Cloud is an option to get started, but you provide the JSON here as it would be for the Docker installation of Hasura. These instructions are unclear for a user on Hasura Cloud.